File: pay-frontend/services/demo-page/src/helpers/payment-sheet.ts
import * as Sdk from '@yandex-pay/sdk/src/typings';

import { getMerchant } from './merchant';

export const getPaymentSheet = (partialSheet: Partial<Sdk.PaymentSheet> = {}): Sdk.PaymentSheet => {
    return {
        ...partialSheet,
        version: 2,
        countryCode: Sdk.CountryCode.Ru,
        currencyCode: Sdk.CurrencyCode.Rub,
        merchant: getMerchant(),
        order: {
            id: 'test-order-id-123',
            total: {
                amount: '40.00',
            },
            items: [
                {
                    label: 'Item A (M)',
                    amount: '15.00',
                },
                {
                    label: 'Item B (Size S)',
                    amount: '25.00',
                },
            ],
        },
        paymentMethods: [
            {
                type: Sdk.PaymentMethodType.Card,
                gateway: 'yandex-trust',
                gatewayMerchantId: 'yandex-market',
                allowedAuthMethods: [Sdk.AllowedAuthMethod.PanOnly],
                allowedCardNetworks: [
                    Sdk.AllowedCardNetwork.Visa,
                    Sdk.AllowedCardNetwork.Mastercard,
                    Sdk.AllowedCardNetwork.Mir,
                    Sdk.AllowedCardNetwork.Uzcard,
                ],
            },
            {
                type: Sdk.PaymentMethodType.Cash,
            },
        ],
        requiredFields: {
            billingContact: {
                email: true,
            },
            shippingContact: {
                name: true,
                email: true,
                phone: true,
            },
            shippingTypes: {
                direct: true,
                pickup: true,
            },
        },
    };
};

File: lib/entities/payment_sheet.py
from dataclasses import dataclass, field
from decimal import Decimal
from enum import Enum, unique
from typing import List, Optional
from uuid import UUID

from pay.lib.entities.enums import AuthMethod, CardNetwork, ClassicPaymentMethodType, PaymentItemType
from pay.lib.entities.order import ContactFields


@unique
class MITOptionsType(Enum):
    RECURRING = 'RECURRING'
    DEFERRED = 'DEFERRED'


@dataclass
class PaymentSheetMITOptions:
    type: MITOptionsType
    optional: bool = False


@dataclass
class PaymentItemQuantity:
    count: Decimal
    label: Optional[str] = None


@dataclass
class PaymentOrderItem:
    amount: Decimal
    label: str
    quantity: Optional[PaymentItemQuantity] = None
    type: Optional[PaymentItemType] = None


@dataclass
class PaymentOrderTotal:
    amount: Decimal
    label: Optional[str] = None


@dataclass
class PaymentOrder:
    id: str
    total: PaymentOrderTotal
    items: Optional[List[PaymentOrderItem]] = None


@dataclass
class PaymentMerchant:
    id: UUID
    name: str
    url: Optional[str] = None


@dataclass
class PaymentMethod:
    method_type: ClassicPaymentMethodType = field(metadata=dict(dump_to='type', load_from='type'))
    gateway: Optional[str] = None
    gateway_merchant_id: Optional[str] = None
    allowed_auth_methods: Optional[List[AuthMethod]] = None
    allowed_card_networks: Optional[List[CardNetwork]] = None
    verification_details: bool = False


@dataclass
class ShippingTypes:
    direct: Optional[bool] = None
    pickup: Optional[bool] = None


@dataclass
class PaymentSheetRequiredFields:
    billing_contact: Optional[ContactFields] = None
    shipping_contact: Optional[ContactFields] = None
    shipping_types: Optional[ShippingTypes] = None


@dataclass
class PaymentSheet:
    version: int
    merchant: PaymentMerchant
    currency_code: str
    country_code: str
    payment_methods: List[PaymentMethod]
    order: PaymentOrder
    required_fields: Optional[PaymentSheetRequiredFields] = None
    mit_options: Optional[PaymentSheetMITOptions] = None
